### ISSUE TITLE
Error at instruction 4: Type java.lang.ClassNotFoundException: org.eolang.benchmark.F not present

### ISSUE THREAD

**Issue by yegor256** [objectionary/jeo-maven-plugin#432]
I'm getting this:

```
org.objectweb.asm.tree.analysis.AnalyzerException: Error at instruction 4: Type java.lang.ClassNotFoundException: org.eolang.benchmark.F not present
	at org.objectweb.asm.tree.analysis.Analyzer.analyze(Analyzer.java:279)
	at org.objectweb.asm.util.CheckClassAdapter.verify(CheckClassAdapter.java:1074)
	at org.eolang.jeo.representation.bytecode.BytecodeClass.bytecode(BytecodeClass.java:171)
	at org.eolang.jeo.representation.xmir.XmlBytecode.bytecode(XmlBytecode.java:97)
	at org.eolang.jeo.representation.EoRepresentation.toBytecode(EoRepresentation.java:103)
	at org.eolang.jeo.improvement.ImprovementBytecodeFootprint.recompile(ImprovementBytecodeFootprint.java:77)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.eolang.jeo.improvement.ImprovementBytecodeFootprint.apply(ImprovementBytecodeFootprint.java:64)
	at org.eolang.jeo.AssembleMojo.execute(AssembleMojo.java:87)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:126)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:328)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:316)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:174)
	at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:75)
	at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:162)
	at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:159)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:73)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:53)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:118)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:261)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:173)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:101)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:906)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:283)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:206)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:283)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:226)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:407)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:348)
Caused by: java.lang.TypeNotPresentException: Type java.lang.ClassNotFoundException: org.eolang.benchmark.F not present
	at org.objectweb.asm.tree.analysis.SimpleVerifier.getClass(SimpleVerifier.java:378)
	at org.objectweb.asm.tree.analysis.SimpleVerifier.isAssignableFrom(SimpleVerifier.java:360)
	at org.objectweb.asm.tree.analysis.SimpleVerifier.isAssignableFrom(SimpleVerifier.java:348)
	at org.objectweb.asm.tree.analysis.SimpleVerifier.isSubTypeOf(SimpleVerifier.java:212)
	at org.objectweb.asm.tree.analysis.BasicVerifier.binaryOperation(BasicVerifier.java:302)
	at org.objectweb.asm.tree.analysis.BasicVerifier.binaryOperation(BasicVerifier.java:44)
	at org.objectweb.asm.tree.analysis.Frame.execute(Frame.java:569)
	at org.objectweb.asm.tree.analysis.Analyzer.analyze(Analyzer.java:171)
	... 33 more
Caused by: java.lang.ClassNotFoundException: org.eolang.benchmark.F
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:496)
	at java.base/java.lang.Class.forName(Class.java:475)
	at org.objectweb.asm.tree.analysis.SimpleVerifier.getClass(SimpleVerifier.java:376)
	... 40 more
[INFO] B.xmir compiled into after/classes/org/eolang/benchmark/B.class (315b)
```

It's hard to understand what file caused the error, but I suspect that it's `B.xmir`:

```
<?xml version="1.0" encoding="UTF-8"?>
<program dob="2024-01-23T04:29:33.009620Z"
         ms="1705984173009"
         name="j$B"
         revision="0.0.0"
         time="2024-01-23T04:29:33.009620Z"
         version="0.0.0">
   <errors/>
   <sheets/>
   <license/>
   <metas>
      <meta>
         <head>package</head>
         <tail>org.eolang.benchmark</tail>
         <part>org.eolang.benchmark</part>
      </meta>
      <meta>
         <head>alias</head>
         <tail>org.eolang.jeo.opcode</tail>
         <part>org.eolang.jeo.opcode</part>
      </meta>
      <meta>
         <head>alias</head>
         <tail>org.eolang.jeo.label</tail>
         <part>org.eolang.jeo.label</part>
      </meta>
   </metas>
   <objects>
      <o abstract="" name="j$B">
         <o base="int" data="bytes" name="access">00 00 00 00 00 00 00 20</o>
         <o base="string" data="bytes" name="supername">6A 61 76 61 2F 6C 61 6E 67 2F 4F 62 6A 65 63 74</o>
         <o base="tuple" name="interfaces" star=""/>
         <o base="field" name="j$f">
            <o base="int" data="bytes">00 00 00 00 00 00 00 12</o>
            <o base="string" data="bytes">4C 6F 72 67 2F 65 6F 6C 61 6E 67 2F 62 65 6E 63 68 6D 61 72 6B 2F 46 3B</o>
            <o base="string" data="bytes"/>
            <o base="string" data="bytes"/>
         </o>
         <o abstract="" name="new">
            <o base="int" data="bytes" name="access">00 00 00 00 00 00 00 00</o>
            <o base="string" data="bytes" name="descriptor">28 4C 6F 72 67 2F 65 6F 6C 61 6E 67 2F 62 65 6E 63 68 6D 61 72 6B 2F 46 3B 29 56</o>
            <o base="string" data="bytes" name="signature"/>
            <o base="tuple" name="exceptions" star=""/>
            <o abstract="" name="arg__Lorg/eolang/benchmark/F;__0"/>
            <o base="seq" name="@">
               <o base="tuple" star="">
                  <o base="label">
                     <o base="string" data="bytes">61 38 35 33 36 36 32 36 2D 62 36 35 65 2D 34 37 31 34 2D 62 64 33 66 2D 65 32 30 32 63 34 39 34 64 30 64 62</o>
                  </o>
                  <o base="opcode" line="999" name="ALOAD-1A">
                     <o base="int" data="bytes">00 00 00 00 00 00 00 19</o>
                     <o base="int" data="bytes">00 00 00 00 00 00 00 00</o>
                  </o>
                  <o base="opcode" line="999" name="INVOKESPECIAL-1B">
                     <o base="int" data="bytes">00 00 00 00 00 00 00 B7</o>
                     <o base="string" data="bytes">6A 61 76 61 2F 6C 61 6E 67 2F 4F 62 6A 65 63 74</o>
                     <o base="string" data="bytes">3C 69 6E 69 74 3E</o>
                     <o base="string" data="bytes">28 29 56</o>
                  </o>
                  <o base="label">
                     <o base="string" data="bytes">36 39 64 63 31 32 31 32 2D 64 63 64 62 2D 34 64 33 35 2D 61 38 31 35 2D 32 63 32 36 65 31 62 38 63 34 38 30</o>
                  </o>
                  <o base="opcode" line="999" name="ALOAD-1C">
                     <o base="int" data="bytes">00 00 00 00 00 00 00 19</o>
                     <o base="int" data="bytes">00 00 00 00 00 00 00 00</o>
                  </o>
                  <o base="opcode" line="999" name="ALOAD-1D">
                     <o base="int" data="bytes">00 00 00 00 00 00 00 19</o>
                     <o base="int" data="bytes">00 00 00 00 00 00 00 00</o>
                  </o>
                  <o base="opcode" line="999" name="PUTFIELD-1E">
                     <o base="int" data="bytes">00 00 00 00 00 00 00 B5</o>
                     <o base="string" data="bytes">6F 72 67 2F 65 6F 6C 61 6E 67 2F 62 65 6E 63 68 6D 61 72 6B 2F 42</o>
                     <o base="string" data="bytes">66</o>
                     <o base="string" data="bytes">4C 6F 72 67 2F 65 6F 6C 61 6E 67 2F 62 65 6E 63 68 6D 61 72 6B 2F 46 3B</o>
                  </o>
                  <o base="label">
                     <o base="string" data="bytes">65 38 36 35 66 37 62 62 2D 63 61 31 35 2D 34 39 65 61 2D 39 65 34 61 2D 38 33 63 64 37 62 34 36 66 36 64 64</o>
                  </o>
                  <o base="opcode" line="999" name="RETURN-1F">
                     <o base="int" data="bytes">00 00 00 00 00 00 00 B1</o>
                  </o>
                  <o base="label">
                     <o base="string" data="bytes">63 30 34 32 37 39 30 36 2D 63 32 61 31 2D 34 36 34 32 2D 38 66 30 31 2D 39 64 35 33 31 31 65 62 31 61 66 32</o>
                  </o>
               </o>
            </o>
         </o>
         <o abstract="" name="j$bar">
            <o base="int" data="bytes" name="access">00 00 00 00 00 00 00 00</o>
            <o base="string" data="bytes" name="descriptor">28 29 49</o>
            <o base="string" data="bytes" name="signature"/>
            <o base="tuple" name="exceptions" star=""/>
            <o base="seq" name="@">
               <o base="tuple" star="">
                  <o base="label">
                     <o base="string" data="bytes">64 62 30 35 32 61 34 34 2D 63 66 38 35 2D 34 30 39 39 2D 38 64 65 35 2D 38 39 61 61 66 38 32 65 62 35 66 39</o>
                  </o>
                  <o base="opcode" line="999" name="ALOAD-20">
                     <o base="int" data="bytes">00 00 00 00 00 00 00 19</o>
                     <o base="int" data="bytes">00 00 00 00 00 00 00 00</o>
                  </o>
                  <o base="opcode" line="999" name="GETFIELD-21">
                     <o base="int" data="bytes">00 00 00 00 00 00 00 B4</o>
                     <o base="string" data="bytes">6F 72 67 2F 65 6F 6C 61 6E 67 2F 62 65 6E 63 68 6D 61 72 6B 2F 42</o>
                     <o base="string" data="bytes">66</o>
                     <o base="string" data="bytes">4C 6F 72 67 2F 65 6F 6C 61 6E 67 2F 62 65 6E 63 68 6D 61 72 6B 2F 46 3B</o>
                  </o>
                  <o base="opcode" line="999" name="INVOKEINTERFACE-22">
                     <o base="int" data="bytes">00 00 00 00 00 00 00 B9</o>
                     <o base="string" data="bytes">6F 72 67 2F 65 6F 6C 61 6E 67 2F 62 65 6E 63 68 6D 61 72 6B 2F 46</o>
                     <o base="string" data="bytes">66 6F 6F</o>
                     <o base="string" data="bytes">28 29 49</o>
                  </o>
                  <o base="opcode" line="999" name="ICONST_2-23">
                     <o base="int" data="bytes">00 00 00 00 00 00 00 05</o>
                  </o>
                  <o base="opcode" line="999" name="IADD-24">
                     <o base="int" data="bytes">00 00 00 00 00 00 00 60</o>
                  </o>
                  <o base="opcode" line="999" name="IRETURN-25">
                     <o base="int" data="bytes">00 00 00 00 00 00 00 AC</o>
                  </o>
                  <o base="label">
                     <o base="string" data="bytes">30 38 34 65 34 37 62 39 2D 31 31 63 34 2D 34 63 33 34 2D 62 33 39 31 2D 34 65 31 66 65 31 30 38 65 38 66 63</o>
                  </o>
               </o>
            </o>
         </o>
      </o>
   </objects>
</program>
```

**Comment by yegor256** [comment]
@volodya-lombrozo full log is here: https://github.com/objectionary/benchmark/actions/runs/7621118403/job/20756983229

**Comment by volodya-lombrozo** [comment]
The problem in compilation of `App.xmir` to bytecode. The source file is `after/generated-sources/opeo-compile/org/eolang/benchmark/App.xmir`. This `App.xmir` file contains the following set of instructions:
```xml
 70                   <o base="label">
 71                      <o base="string" data="bytes">66 36 37 36 31 39 63 39 2D 63 36 38 30 2D 34 37 36 62 2D 61 38 30 39 2D 30 65 34 30 38 64 38 32 39 64 37 66</o>
 72                   </o>
 73                   <o base="opcode" line="999" name="NEW-27">
 74                      <o base="int" data="bytes">00 00 00 00 00 00 00 BB</o>
 75                      <o base="string" data="bytes">6F 72 67 2F 65 6F 6C 61 6E 67 2F 62 65 6E 63 68 6D 61 72 6B 2F 42 41</o>
 76                   </o>
 77                   <o base="opcode" line="999" name="DUP-28">
 78                      <o base="int" data="bytes">00 00 00 00 00 00 00 59</o>
 79                   </o>
 80                   <o base="opcode" line="999" name="BIPUSH-29">
 81                      <o base="int" data="bytes">00 00 00 00 00 00 00 10</o>
 82                      <o base="int" data="bytes">00 00 00 00 00 00 00 2A</o>
 83                   </o>
 84                   <o base="opcode" line="999" name="INVOKESPECIAL-2A">
 85                      <o base="int" data="bytes">00 00 00 00 00 00 00 B7</o>
 86                      <o base="string" data="bytes">6F 72 67 2F 65 6F 6C 61 6E 67 2F 62 65 6E 63 68 6D 61 72 6B 2F 42 41</o>
 87                      <o base="string" data="bytes">3C 69 6E 69 74 3E</o>
 88                      <o base="string" data="bytes">28 29 56</o>
 89                   </o>
 90                   <o base="opcode" line="999" name="INVOKEVIRTUAL-2B">
 91                      <o base="int" data="bytes">00 00 00 00 00 00 00 B6</o>
 92                      <o base="string" data="bytes">6F 72 67 2F 65 6F 6C 61 6E 67 2F 62 65 6E 63 68 6D 61 72 6B 2F 42</o>
 93                      <o base="string" data="bytes">62 61 72</o>
 94                      <o base="string" data="bytes">28 29 49</o>
 95                   </o>
 96                   <o base="opcode" line="999" name="IRETURN-2C">
 97                      <o base="int" data="bytes">00 00 00 00 00 00 00 AC</o>
 98                   </o>
 99                   <o base="label">
100                      <o base="string" data="bytes">63 33 63 61 61 30 31 32 2D 65 31 37 38 2D 34 64 35 34 2D 39 61 34 38 2D 38 63 35 37 64 35 65 34 35 34 39 38</o>
101                   </o>
```

If we convert the xml above to a human-readable list of bytecode instructions we get:
```
0 NEW "org/eolang/benchmark/BA"
1 DUP
2 BIPUSH 42
3 INVOKESPECIAL "org/eolang/benchmark/BA" "<init>" "()V"
```  

As you can see the last argument of `INVOKESPECIAL` is "()V", which seems wrong. Probably it should be "(I)V". 
In other words, now we have `new BA()`, but we want `new BA(I)`.

**Comment by yegor256** [comment]
@volodya-lombrozo thus, it's a bug in the opeo-maven-plugin?

**Comment by volodya-lombrozo** [comment]
@yegor256 Yes, I will fix it soon. As for:
> It's hard to understand what file caused the error,

Can you update `jeo-maven-plugin` to `0.2.20` version, please? Now it should fail immediately if bytecode check fails

**Comment by yegor256** [comment]
@volodya-lombrozo [updated](https://github.com/objectionary/benchmark/commit/8cfdb18319c30f3d6c63607317e8bb1cd9f19a5f), indeed it fails now, thanks

**Comment by yegor256** [comment]
@volodya-lombrozo submitted to opeo, closing this one here: https://github.com/objectionary/opeo-maven-plugin/issues/110
